src/rollback_to_stable/rts_btree.c

```diff
@@ -107,6 +107,17 @@ __rts_btree_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *firs
                     F_CLR(stable_upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS);
                 if (tombstone != NULL)
                     F_CLR(tombstone, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS);
+            } else if (WT_IS_HS(session->dhandle) && stable_upd->type != WT_UPDATE_TOMBSTONE) {
+                /*
```

Comment:
I don't think this code segment will ever be executed.
WT_IS_HS(...) means this is executed only for History store.
This is under the condition `F_ISSET(stable_upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS)` from line 77, which will be true only for data store.
@kommiharibabu, let me know if I am missing something

Reply:
@bvpvamsikrishna Good finding. Yes, that's correct. The `else if` must be with `F_ISSET(stable_upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS)` instead of `if (!dryrun) {`.
@ruby-oujo Please adjust the check.

Reply:
Good point, addressed here: [ae899d3](https://github.com/wiredtiger/wiredtiger/pull/8969/commits/ae899d3897562f4c94c7a8b1cf60b6273a4be5f3)